node/lifecycle_test.go

```diff
@@ -446,14 +474,15 @@ func testUpdatePodWhileRunningScenario(ctx context.Context, t *testing.T, s *sys
 	// Setup a watch to check if the pod is in running
 	watcher, err := s.client.CoreV1().Pods(testNamespace).Watch(listOptions)
 	assert.NilError(t, err)
+	defer watcher.Stop()
 	go func() {
-		_, watchErr := watchutils.UntilWithoutRetry(ctx, watcher,
+		newPod, watchErr := watchutils.UntilWithoutRetry(ctx, watcher,
 			// Wait for the pod to be started
 			func(ev watch.Event) (bool, error) {
 				pod := ev.Object.(*corev1.Pod)
 				return pod.Status.Phase == corev1.PodRunning, nil
 			})
-
+		p = newPod.Object.(*corev1.Pod).DeepCopy()
```

Comment:
Does DeepCopy grab a lock?

Reply:
No, it doesn't need to. The pod objects returned by the fake client-go  should be safe to *read* (but not neccessarily write) to.

Reply:
But why is this needed in the first place?